src/AffectedUserFlow/CodeInput/CodeInputForm.tsx

```diff
@@ -54,16 +63,21 @@ const CodeInputForm: FunctionComponent = () => {
   const [errorMessage, setErrorMessage] = useState(defaultErrorMessage)
   const [isFocused, setIsFocused] = useState(false)
 
-  const codeLengthMin = 6
-  const codeLengthMax = 16
-  const codeIsInvalidLength =
-    code.length < codeLengthMin || code.length > codeLengthMax
-  const codeContainsNonAlphanumericChars = (code: string) =>
-    !code.match(/^[a-zA-Z0-9]*$/)
+  const linkCode: string | undefined = route?.params?.c || route?.params?.code
+  useEffect(() => {
+    if (linkCode) {
+      setCode(linkCode)
```

Comment:
@devinjameson , ended up moving to this as the `API.postCode` call uses the component state for the code and keeping a separate `linkCode` and `userInputedCode` made that somewhat awkward.  I think this is another vote for the make a bespoke screen/form for this use case, but this seems to work with the minimum amount of new logic.